File: c#/src/Docmost.Sdk/Models/Attachment.cs
namespace Docmost.Sdk.Models;

public class Attachment
{
    [System.Text.Json.Serialization.JsonPropertyName("id")]
    public string? Id { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("fileName")]
    public string? FileName { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("fileExt")]
    public string? FileExt { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("filePath")]
    public string? FilePath { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("fileSize")]
    public string? FileSize { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("mimeType")]
    public string? MimeType { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("type")]
    public string? Type { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("pageId")]
    public string? PageId { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("creatorId")]
    public string? CreatorId { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("spaceId")]
    public string? SpaceId { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("workspaceId")]
    public string? WorkspaceId { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("createdAt")]
    public string? CreatedAt { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("updatedAt")]
    public string? UpdatedAt { get; set; }

}

File: go/docmost/models/attachment.go
package models

// Attachment from OpenAPI.
type Attachment struct {
	Id *string `json:"id,omitempty"`
	FileName *string `json:"fileName,omitempty"`
	FileExt *string `json:"fileExt,omitempty"`
	FilePath *string `json:"filePath,omitempty"`
	FileSize *string `json:"fileSize,omitempty"`
	MimeType *string `json:"mimeType,omitempty"`
	Type_ *string `json:"type,omitempty"`
	PageId *string `json:"pageId,omitempty"`
	CreatorId *string `json:"creatorId,omitempty"`
	SpaceId *string `json:"spaceId,omitempty"`
	WorkspaceId *string `json:"workspaceId,omitempty"`
	CreatedAt *string `json:"createdAt,omitempty"`
	UpdatedAt *string `json:"updatedAt,omitempty"`
}
